client-side-encryption.md

```diff
@@ -0,0 +1,856 @@
+- **Cost Impact**: Reduces encryption costs from 63% to just 0.2-0.3% of revenue
+- **Revenue Generation**: Creates new revenue streams (for example: +$5-20/month per user who opts in)
+
+**Key Strategic Benefits**:
+
+- **Sustainable Economics**: Only users who value encryption pay for it, eliminating cost burden
+- **Customer Retention**: No disruption to existing workflows - all users continue with current storage patterns
+- **Conversion Driver**: Privacy-conscious free users must upgrade to access encryption features
+- **Mainstream Accessibility**: Email OTP and social authentication eliminate crypto wallet barriers when encryption is chosen
+- **Enterprise Ready**: Multiple compliance tiers capture enterprise market requirements
+- **Revenue Growth**: Encryption becomes profitable feature rather than cost center
+
+## 2. Proposed Enhancements
+
+This RFC addresses the identified limitations through a strategy that maintains the security properties of the existing system while improving usability, reliability, and enterprise sales opportunities. The proposal introduces **seven key improvements**:
+
+### 2.1 Hybrid Architecture: Lit Protocol + Google KMS
+
+Implement a flexible dual-path encryption system that provides **multiple tiers of security and backup options**:
+
+#### 2.1.1 Starter (Free) Users: No Encryption
+
+- **Default**: Standard unencrypted storage only
+- **Encryption**: Not available - requires paid plan upgrade
+- **Rationale**: Encryption is a premium feature that drives subscription conversions
+- **Use Case**: Basic storage needs without privacy requirements
+- **Upgrade Incentive**: Privacy-conscious users must subscribe to access encryption features
+
+#### 2.1.2 Paid Users (Lite/Business/Enterprise): Encryption-as-Premium-Feature
+
+- **Default**: Standard unencrypted storage (preserves existing customer workflows)
+- **Optional Encryption Tiers**:
+  - **Privacy Tier**: Lit Protocol end-to-end encryption
+  - **Compliance Tier**: Google KMS with certifications
+- **Use Case**: Paid users who need specific privacy/compliance requirements without disrupting standard workflows
+- **Public Sharing**: Unencrypted files remain easily shareable and collaborative
+
+### 2.2 Lit PKP Integration
+
+Add support for Lit Protocol's Programmable Key Pairs (PKPs) alongside existing wallet-based authentication. PKPs enable social login providers (Google, Discord, WebAuthn, Stytch, etc) for users who prefer not to manage Ethereum wallets and private keys, while maintaining backward compatibility with existing wallet-based workflows.
+
+> Programmable Key Pairs (PKPs) are ECDSA public/private key pairs created by the Lit network using Distributed Key Generation (DKG). Each Lit node holds a share of the private key, and more than two-thirds of these shares must be collected to execute a given action (i.e. signing a transaction).
+> ref: https://developer.litprotocol.com/user-wallets/pkps/overview
+
```

Comment:
I think this definition a bit confused, since in the docs it's also said that PKPs are NFTs.
So I would also add this:

> The PKP is represented as an NFT, and whoever owns this NFT controls the key pair and can set custom signing logic or access rules.
> ref: https://developer.litprotocol.com/user-wallets/pkps/quick-start#mint-a-pkp-and-add-permitted-scopes